api-lib/insertCoLinksGive.ts

```diff
@@ -89,3 +94,25 @@ export const insertCoLinksGive = async (
 
   return { giveId, newPoints };
 };
+
+export const publishCastGiveDelivered = async (
+  hash: string,
+  giveId: number,
+  precache = true
+) => {
+  if (precache) {
+    // PRE-CACHE farme and image by calling the URL
+    await fetch(getFrameUrl('give', giveId), {
+      signal: AbortSignal.timeout(10000),
+    });
```

Comment:
will make handler resposne slwoer, but no other easy way to precache the frame image.